superqt/combobox/_enum_combobox.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional, Type, TypeVar, Union
+from enum import Enum
+
+from ..qtcompat.QtWidgets import QComboBox
+from ..qtcompat.QtCore import Signal, PYQT5
+
+EnumType = TypeVar("EnumType", bound=Enum)
+enum_type = Enum if PYQT5 else object
+
+def _get_name(enum_value: Enum):
+    """Create human readable name if user does not provide own implementation of __str__"""
+    if enum_value.__str__.__module__ != "enum":
+        # check if function was overloaded
+        name = str(enum_value)
+    else:
+        name = enum_value.name.replace("_", " ")
+    return name
+
+
+class EnumComboBox(QComboBox):
+    """
+    This is implementation of variant of EnumComboBox for select enum value.
+    If enum class does not provide own __str__ implementation then human readable name is provide using name of item
+    replacing underscores with spaces.
+    """
+    currentValueChanged = Signal(enum_type)
+
+    def __init__(self, parent=None, enum:Optional[Type[EnumType]] = None):
+        super().__init__(parent)
+        self._enum = enum
+        if enum is not None:
+            self.setEnum(enum)
+        self.currentIndexChanged.connect(self._emit_signal)
+
+    def setEnum(self, enum: Type[EnumType]):
+        """
+        Set enum class from which members value should be selected
+        """
+        self.clear()
+        self._enum = enum
+        super().addItems(list(map(_get_name, enum.__members__.values())))
+
+    def enum(self) -> Optional[Type[EnumType]]:
+        return self._enum
+
+    def clear(self):
+        self._enum = None
+        super().clear()
+
+    def currentValue(self) -> EnumType:
+        """current value as Enum member"""
+        if self._enum is None:
+            raise RuntimeError("Enum value is None")
```

Comment:
thoughts on just returning `None` rather than raising an Error?

Reply:
I think that in the future this widget should also support Optional annotation (I do not know who is the best option to check Optional for now). 

So I prefer exception to fast inform User that he does not properly initialize widget and use option which is distinguishable from None selected.

Reply:
hm... maybe let's leave that for when we add an optional (null) member?  ... or go ahead and add it here?  I feel like an exception in something like `currentX` is rather unusual in Qt... normally it would give you either None or some sort of null value like `-1`

Reply:
for instance, here's an "uninitialized" QComboBox... no exceptions raised:
```py
In [12]: from qtpy.QtWidgets import QComboBox

In [13]: q = QComboBox()

In [14]: q.currentIndex()
Out[14]: -1

In [15]: q.currentText()
Out[15]: ''

In [16]: q.currentData()
```